p2p/host/relay/autorelay.go

```diff
@@ -124,28 +139,38 @@ func (ar *AutoRelay) background(ctx context.Context) {
 	}
 }
 
-func (ar *AutoRelay) findRelays(ctx context.Context) bool {
-	if ar.numRelays() >= DesiredRelays {
-		return false
-	}
+func (ar *AutoRelay) findRelays(ctx context.Context, startFindRelaysc <-chan bool, updatec chan<- bool) {
+	running := false
 
-	update := false
-	for retry := 0; retry < 5; retry++ {
-		if retry > 0 {
-			log.Debug("no relays connected; retrying in 30s")
-			select {
-			case <-time.After(30 * time.Second):
-			case <-ctx.Done():
-				return update
-			}
+	checkUpdate := func(ctx context.Context) {
```

Comment:
can you declare this as a method on AutoRelay, rather than an anonymous function?